### ISSUE TITLE
Improve parameter documentation

### ISSUE THREAD

**Issue by bbockelm** [PelicanPlatform/pelican#578]
The [parameter documentation page](https://docs.pelicanplatform.org/parameters) could use some improving.

0.  Provide a search box at the top where people can live filter for parameters they are interested in.  Should contain buttons that filter quickly on a specific component.
1. Make each parameter description more of a self-contained card.
2.  Add visual cues for the different components instead of a rendered list.  Similarly, add some structure around the type and default.  The type should be rendered human friendly ("integer", not "int") with tooltips
3. Might play around with having the sub-parameters in a collapsable structure so you can easily suppress ones you're not interested in.  Not sure if this would render well.
4. Add a sentence to the top describing what the page is.

**Comment by CannonLock** [comment]
@Rhymlore For # 2 go right into the parameters file and make it more friendly there rather then casting it. 

https://github.com/PelicanPlatform/pelican/blob/main/docs/parameters.yaml

int -> integer
duration -> time.Duration
bool -> boolean

**Comment by CannonLock** [comment]
@rhymlore # 1 Use the Autocomplete component: 

If I search Origin, below it will show a list like:

- Origin.EnableIssuer
- Origin.ScitokensRestrictedPaths

https://mui.com/material-ui/react-autocomplete/#search-as-you-type

**Comment by CannonLock** [comment]
Remaining after #679 is merged will be the buttons to filter quickly by component.

**Comment by CannonLock** [comment]
Put the final TODO in a new ticket: https://github.com/PelicanPlatform/pelican/issues/742
